packages/webpack/src/plugins/webpack-nx-build-coordination-plugin.ts

```diff
@@ -56,44 +45,55 @@ export class WebpackNxBuildCoordinationPlugin {
     this.currentlyRunning = 'nx-build';
     return new Promise<void>((res) => {
```

Comment:
Would changing this to `return await new Promise ...` work without having to put `this.currentlyRunning = 'none';` in the `.then()` callback? I think previously the promise is not being awaited, thus the `finally` is called before the promise resolves.

Generally returning promises should always be awaited on, or else you get problems like this, and also stacktraces are messed up.

Reply:
Yeah it makes no difference. 

We can add the `await` anyway if you think it would be better to do so, but the `finally` is happening for the `try-catch-finally` block _within_ the Promise.

The code inside that promise is all non-blocking code, so it all runs through, including the `finally`.

Reply:
Huh, that's surprising, the `finally` should fire after promise resolves normally, even after the `.then()`. Maybe it's the way the code is being transpiled by tsc? Either case, I guess it's fine in `.then()` if it's working.